libs2ecore/src/S2EExecutor.cpp

```diff
@@ -1394,7 +1407,16 @@ S2EExecutor::StatePair S2EExecutor::fork(ExecutionState &current, const klee::re
         currentState->forkDisabled = true;
     }
 
-    res = Executor::fork(current, condition, keepConditionTrueInCurrentState);
+    if (condition) {
+        res = Executor::fork(current, *condition, keepConditionTrueInCurrentState);
+    } else {
+        // The caller has not passed any condition to us,
+        // so we'll create a dummy condition and pass it to klee::Executor::fork().
+        klee::ref<klee::Expr> dummyVar = currentState->createSymbolicValue("", klee::Expr::Bool);
+        klee::ref<klee::Expr> dummyCond =
+            klee::EqExpr::create(dummyVar, klee::ConstantExpr::create(false, klee::Expr::Bool));
+        res = Executor::fork(current, dummyCond, false);
```

Comment:
Actually, you can create an overloaded Executor::fork() version that is like the original fork but without any path constraint code. It would just clone the state essentially.

Reply:
Thanks for your help, Vitaly! I'll look into `Executor::fork()`.